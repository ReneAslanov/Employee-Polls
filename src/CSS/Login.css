.login-wrapper{
    display: flex;
    flex-direction: column;

}

.login-label{
    margin: 0 auto;
    font-size: 1.3rem;
    padding: 1rem;
}

.login-headline{
    margin: 0 auto;
}

.login-form{
    margin: 2rem auto;
    display: flex;
    justify-content: center;
    flex-direction: column;
    max-width: 600px;
}

.login-input{
    height: 30px;
    border: 1px solid #e1e5eb;
}

.login-submit{
    width: 200px;
    margin: 20px auto;
    border: 1px solid #e1e5eb;
    background-color: #fff;
    padding: 1rem;
    cursor: pointer;
    text-align: center;
    text-decoration: none;
    color: #b3b8c0;
    transition: border-color 1s cubic-bezier(0.075, 0.82, 0.165, 1), color 1s cubic-bezier(0.075, 0.82, 0.165, 1);
}

.login-submit:hover{
    border-color: #000;
    color: #000;
}