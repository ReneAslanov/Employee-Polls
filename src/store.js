import { legacy_createStore as createStore} from "redux";
import middleware from './middleware';
import reducer from "./reducers/index";

const store = createStore(reducer, middleware);

export default store;

