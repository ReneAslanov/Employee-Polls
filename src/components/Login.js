import "../CSS/Login.css"
import { Link, useNavigate } from "react-router-dom";
import {logoutUser, setAuthedUser} from "../actions/authedUser";
import {connect, useDispatch} from "react-redux";
import {useState} from "react";

function Login(props)
{
    const [username, setUsername] = useState("");
    const [password, setPassword] = useState("");
    const navigate = useNavigate();
    const dispatch = useDispatch();

    function validUser(e)
    {
        e.preventDefault();

        const isUser = props.users.find(obj => {
            if(obj.name === username && obj.password === password)
            {
                dispatch(setAuthedUser(obj));
                navigate("/home");
                return true;
            }
            return undefined
        })

        if(isUser === undefined)
        {
            dispatch(logoutUser());
            document.getElementById("user").value = "";
            document.getElementById("password").value = "";
            alert("Username or Password is incorrect, pls enter valid Username and Password");
        }
    }

    return(
        <form className="login-form" id="login-form">
            <h1 className="login-headline">Employee Polls</h1>
            <div className="login-wrapper">
                <label htmlFor="user" className="login-label">User</label>
                <input type="text" placeholder="Username" name="User" id="user" data-testid="user" className="login-input" onChange={event => setUsername(event.target.value.trim())}></input>

                <label htmlFor="password" className="login-label">Password</label>
                <input type="password" placeholder="Password" name="Password" id="password" data-testid="password" className="login-input" onChange={event => setPassword(event.target.value.trim())}></input>
            </div>


            <Link to="/home" className="login-submit" onClick={validUser} id="home-link">submit</Link>
        </form>
    )
}

function mapStateToProps({users, questions})
{
    return{
        users: Object.values(users),
        questions
    }
}

export default connect(mapStateToProps)(Login);