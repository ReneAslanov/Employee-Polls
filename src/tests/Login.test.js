import Login from "../components/Login";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../store";
import { BrowserRouter } from "react-router-dom";

describe("Login", () => {
    it("should render the Login component properly", () => {
        let view = render(
        <Provider store={store}>
            <BrowserRouter>
                <Login />
            </BrowserRouter>
        </Provider>);

        const userName = screen.getByTestId("user");
        const password = screen.getByTestId("password");
        expect(view).toMatchSnapshot();
        expect(userName).toBeInTheDocument();
        expect(password).toBeInTheDocument();
    })
})