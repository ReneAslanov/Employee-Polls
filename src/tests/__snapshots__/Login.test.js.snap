// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the Login component properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="login-form"
        id="login-form"
      >
        <h1
          class="login-headline"
        >
          Employee Polls
        </h1>
        <div
          class="login-wrapper"
        >
          <label
            class="login-label"
            for="user"
          >
            User
          </label>
          <input
            class="login-input"
            data-testid="user"
            id="user"
            name="User"
            placeholder="Username"
            type="text"
          />
          <label
            class="login-label"
            for="password"
          >
            Password
          </label>
          <input
            class="login-input"
            data-testid="password"
            id="password"
            name="Password"
            placeholder="Password"
            type="password"
          />
        </div>
        <a
          class="login-submit"
          href="/home"
          id="home-link"
        >
          submit
        </a>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="login-form"
      id="login-form"
    >
      <h1
        class="login-headline"
      >
        Employee Polls
      </h1>
      <div
        class="login-wrapper"
      >
        <label
          class="login-label"
          for="user"
        >
          User
        </label>
        <input
          class="login-input"
          data-testid="user"
          id="user"
          name="User"
          placeholder="Username"
          type="text"
        />
        <label
          class="login-label"
          for="password"
        >
          Password
        </label>
        <input
          class="login-input"
          data-testid="password"
          id="password"
          name="Password"
          placeholder="Password"
          type="password"
        />
      </div>
      <a
        class="login-submit"
        href="/home"
        id="home-link"
      >
        submit
      </a>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
